redux/warGameSlice.test.ts

```diff
@@ -0,0 +1,118 @@
+import reducer, { setPlayerReady, startRound } from "./warGameSlice";
+
+const initialState = {
+  cardListPlayerOne: [
+    { answer: 0, question: "" },
+    { answer: 1, question: "" },
+    { answer: 2, question: "" },
+  ],
+  cardListPlayerTwo: [
+    { answer: 3, question: "" },
+    { answer: 4, question: "" },
+    { answer: 5, question: "" },
+  ],
+  currentRoundIndex: 0,
+  playerOneReady: false,
+  playerTwoReady: false,
+};
+
+test("should return the initial state", () => {
+  expect(reducer(undefined, { type: "no action" })).toEqual(initialState);
+});
+
+test("test player one ready action", () => {
+  expect(reducer(initialState, setPlayerReady(1))).toEqual({
+    cardListPlayerOne: [
+      { answer: 0, question: "" },
+      { answer: 1, question: "" },
+      { answer: 2, question: "" },
+    ],
+    cardListPlayerTwo: [
+      { answer: 3, question: "" },
+      { answer: 4, question: "" },
+      { answer: 5, question: "" },
+    ],
+    currentRoundIndex: 0,
+    playerOneReady: true,
+    playerTwoReady: false,
+  });
+});
+
+test("test player two ready action", () => {
+  expect(reducer(initialState, setPlayerReady(2))).toEqual({
+    cardListPlayerOne: [
+      { answer: 0, question: "" },
+      { answer: 1, question: "" },
+      { answer: 2, question: "" },
+    ],
+    cardListPlayerTwo: [
+      { answer: 3, question: "" },
+      { answer: 4, question: "" },
+      { answer: 5, question: "" },
+    ],
+    currentRoundIndex: 0,
+    playerOneReady: false,
+    playerTwoReady: true,
+  });
+});
+
+test("test both players ready and start round action", () => {
+  const firstState = reducer(initialState, setPlayerReady(1));
+  const secondState = reducer(firstState, setPlayerReady(2));
+  expect(reducer(secondState, startRound(null))).toEqual({
+    cardListPlayerOne: [
+      { answer: 0, question: "" },
```

Comment:
if we are starting a round and assigning values for playerOneCurrentCard and playerTwoCurrentCard, should we remove those cards from the respective card lists? Or should we keep them there for now and remove them later. Remember at the end of the round, the winning player needs to take both cards and put it at the bottom of their card list.

Reply:
I believe we should keep them there for now and then have another function to remove once they have been assigned and evaluated, just for having them in separate functions